### PR TITLE
Aave governance v2 adapter

### PR DESCRIPTION
**Pull request by jareknowotka**
Proposal for AAVE Governance V2 adapter, I see there is already one great submission from Noah. This one is quite similar with one difference around testing. It's using a mocked AAVE V2 token. It reduces some complexity on the testing, however set issuance isn't mocked, because it wasn't in scope for the adapter.

*Notes:*

* V2 introduced possibility to delegate two types of power: voting power and proposal creation power. By default `getDelegateCalldata` delegates both in one call. If we would like to choose what to delegate we would need to add a new param the `getDelegateCalldata` function.

https://docs.aave.com/developers/protocol-governance/governance#delegatebytype
https://github.com/aave/aave-token-v2/blob/master/contracts/interfaces/IGovernancePowerDelegationToken.sol#L6

### CODE REVIEW

File: contracts/protocol/integration/AaveGovernanceV2Adapter.sol [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/*
+    Copyright 2020 Set Labs Inc.
```
Comment (richardliang) [inline contracts/protocol/integration/AaveGovernanceV2Adapter.sol:2]: *2021

File: contracts/protocol/integration/AaveGovernanceV2Adapter.sol [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+pragma solidity 0.6.10;
+pragma experimental "ABIEncoderV2";
+
+
+/**
+ * @title AaveGovernanceV2Adapter
+ * @author Set Protocol
+ *
+ * Governance adapter for Aave governance V2 that returns data for voting
```
Comment (richardliang) [inline contracts/protocol/integration/AaveGovernanceV2Adapter.sol:27]: voting, delegation of votes, creation of proposals

File: contracts/protocol/integration/AaveGovernanceV2Adapter.sol [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+ */
+contract AaveGovernanceV2Adapter {
+
+    /* ============ Constants ============ */
+
+    // // 1 is a vote for in AAVE
```
Comment (richardliang) [inline contracts/protocol/integration/AaveGovernanceV2Adapter.sol:33]: Can remove commented out code

File: contracts/protocol/integration/AaveGovernanceV2Adapter.sol [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    /**
+     * Reverts as AAVE currently does not have a register mechanism in governance
+     */
+    function getRevokeCalldata() external view returns (address, uint256, bytes memory) {
+        revert("To revoke delegation, set the delegatee address to the delegator address");
+    }
+
+    /**
+     * Generates the calldata to create a new proposal.
+     * The caller must have proposition power higher than PROPOSITION_THRESHOLD to create a proposal.
```
Comment (richardliang) [inline contracts/protocol/integration/AaveGovernanceV2Adapter.sol:112]: Really like the level of detail in the description here!

**Review by richardliang — COMMENTED** [review]
In general, really like the level of detail in the comments, actually made these suggestions to Noah's PR. The functionality is pretty similar here, so may just merge Noah's just because I started reviewing his earlier

**Comment by cgewecke** [comment]
Hi @jareknowotka. Thanks so much for this. In the end, Noah's Governance adapter was merged. Am going to close here. 

As an aside - in case you have other PRs you'd like to submit - this repo is being deprecated in favor of https://github.com/SetProtocol/set-protocol-v2. See you over there :)

**Pull request closed without merging**